app/src/main/java/com/example/househomey/SignUpFragment.java

```diff
@@ -0,0 +1,268 @@
+        usernameEdittext = rootView.findViewById(R.id.signup_username);
+        emailEdittext = rootView.findViewById(R.id.signup_email);
+        passwordEdittext = rootView.findViewById(R.id.signup_password);
+        confirmPasswordEdittext = rootView.findViewById(R.id.signup_confirm_password);
+        signupButton = rootView.findViewById(R.id.signup_button);
+
+        TextView signinRedirectMessage = rootView.findViewById(R.id.signin_redirect_message);
+        TextView signinRedirect = rootView.findViewById(R.id.signin_redirect);
+
+        signupButton.setText("Register");
+        signinRedirectMessage.setText("Already have an account?");
+        signinRedirect.setText(R.string.sign_in_string);
+
+        signupButton.setOnClickListener(v -> {
+            signupButton.setBackgroundResource(R.drawable.signin_button_clicked);
+            signupButton.setTextColor(getResources().getColor(R.color.creme, rootView.getContext().getTheme()));
+            new Handler(Looper.getMainLooper()).postDelayed(() -> {
+                signupButton.setBackgroundResource(R.drawable.signin_button_unclicked);
+                signupButton.setTextColor(getResources().getColor(R.color.brown, rootView.getContext().getTheme()));
+            }, 150);
+
+            collRef = FirebaseFirestore.getInstance().collection("user");
+            if (confirmPassword()) {
+                collRef.document(username).get().addOnSuccessListener(task -> {
+                    if (task.getData() == null) {
+                        UserProfileChangeRequest profileChangeRequest =
+                                new UserProfileChangeRequest.Builder().setDisplayName(username).build();
+                        auth.createUserWithEmailAndPassword(email, password).addOnCompleteListener(task1 -> {
+                            if (task1.isSuccessful()) {
+                                FirebaseUser user = auth.getCurrentUser();
+                                if (user == null) {
+                                    emailEdittext.setError("database error");
+                                    usernameEdittext.setError("database error");
+                                } else {
+                                    user.updateProfile(profileChangeRequest).addOnCompleteListener(task12 -> {
+                                        if (task12.isSuccessful()) {
+                                            Map<String, Object> fields = new HashMap<>();
+                                            fields.put("email", user.getEmail());
+                                            collRef.document(user.getDisplayName()).set(fields).addOnCompleteListener(task2 -> {
+                                                if (task2.isSuccessful()) {
+                                                    // Replace YourSignUpFragment with the actual class name of your sign-up fragment
+                                                    FragmentManager fragmentManager = getParentFragmentManager();
```

Comment:
use the navigateToFragment method instead of this block

Reply:
done